packages/core/src/components/Form/Form.tsx

```diff
@@ -643,6 +643,11 @@ export class Form<FormComponents extends ValueMap = {}> extends React.Component<
                 },
             },
         });
+
+        // Force validation if component is not initially pristine to populate validationData
+        if (!componentRef.isPristine()) {
```

Comment:
This is a good solution. It was unaware that the validaiton logic wasn't being executed on mount if the component isn't pristine. 

Thanks for making this change 🙂